Implement the Google Product Category JS handler change method

wvega opened the pull request:
# Summary

This PR adds the handle for the `change` event of the Google Product Category select fields.

### Story: [CH 63610](https://app.clubhouse.io/skyverge/story/63610)
### Release: #1477 

## Details

The PR also updates the constructor to create the necessary select field to represent the previously selected hierarchy.

## UI Changes

- Select fields: https://cloud.skyver.ge/mXuye1z0

## QA

### Setup

1. Add `add_filter( 'wc_facebook_commerce_is_available', '__return_true' )`
1. Add `add_filter( 'wc_facebook_commerce_is_connected', '__return_true' )`

### Steps

#### Select left category

1. Go to WooCommerce > Facebook > Instagram Checkout
    - [x] Select `Arts & Entertainment` in the first select field
    - [x] Select `Hobbies & Creative Arts` in the second field
    - [x] Select `Arts & Crafts` in the third field
    - [x] Select `Arts & Crafts Kits` in the third field
    - [x] Select `Drawing & Painting Kits` in the third field
1. Save changes
    - [x] All the previously selected categories are automatically selected
    - [x] The `wc_facebook_google_product_category_id` option is set to `4778`

#### Select mid-level category

1. Go to WooCommerce > Facebook > Instagram Checkout
    - [x] Select `Arts & Entertainment` in the first select field
    - [x] Select `Hobbies & Creative Arts` in the second field
    - [x] Select `Arts & Crafts` in the third field
    - [x] The next select shows up
1. Save changes
    - [x] All the previously selected categories are automatically selected
    - [x] The next select shows up
    - [x] You can select `Arts & Crafts Kits` in the third field

#### Clear a previously selected value

1. Go to WooCommerce > Facebook > Instagram Checkout
    - [x] Select `Arts & Entertainment` in the first select field
    - [x] Select `Hobbies & Creative Arts` in the second field
    - [x] Select `Arts & Crafts` in the third field
1. Click the x icon next to `Hobbies & Creative Arts`
1. Click Save changes
    - [x] `Arts & Entertainment` is the selected category
    - [x] The next select shows up
    - [x] You can select `Hobbies & Creative Arts` in the second field
1. Click the x icon next to `Arts & Entertainment`
    - [x] The UI returns to the initial state with no categories selected
1. Click Save changes
    - [x] The UI remains in the initial state with no categories selected

## Before merge

- [ ] I have confirmed these changes in each supported minor WooCommerce version

ChaseWiseman reviewed (changes requested):
@wvega nicely done! It's cool to see this in action.

I only notice one small problem:

- [ ] When only selecting as far as a mid-level category and saving, the form is loaded without the option to select more specific categories after it. Steps to replicate:
    1. Select "Cameras & Optics"
    1. Select "Photography" but stop there
        - The next select should show: https://cloud.skyver.ge/12uJWvzz
    1. Save changes
        - The next select is not shown: https://cloud.skyver.ge/Apuk49Dg
        - You can select a different option and reselect the desired option to bring the next select back

It seems like we may need to account for options in `addInitialSelects()`, or maybe trigger a change event somewhere.

wvega commented:
@ChaseWiseman thank you for catching that issue and the steps to reproduce it!

I added changes to render an extra select field if the selected category has options (be09c27) and made it possible to clear the selected categories at any level using Select2's `allowClear`: f07747e and 0b643ba.

I also updated the QA instructions with steps to cover the issue you found and the allow clear feature.
